homeassistant/components/vesync/common.py

```diff
@@ -0,0 +1,87 @@
+"""Common utilities for VeSync Component."""
+import logging
+import asyncio
+from datetime import timedelta
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'vesync'
+
+CONF_SWITCHES = 'switches'
+
+
+async def async_process_devices(hass, manager):
+    """Assign devices to proper component."""
+    devices = {}
+    devices[CONF_SWITCHES] = []
+
+    await hass.async_add_executor_job(manager.update)
+
+    if manager.outlets:
+        devices[CONF_SWITCHES].extend(manager.outlets)
+        _LOGGER.info("%d VeSync outlets found", len(manager.switches))\
+
+    reg_switch = 0
+    if manager.switches:
+        for switch in manager.switches:
+            if not switch.is_dimmable():
+                devices[CONF_SWITCHES].append(switch)
+                reg_switch += 1
+        if reg_switch > 0:
+            _LOGGER.info("%d VeSync standard switches found", reg_switch)
+
+    return devices
+
+
+async def async_add_entities_retry(hass,
```

Comment:
I was referring the the TP-Link component, I thought it might be useful if there is an exception while adding device.  I think that was more of a local issue with the tplink platform, I removed it